bdk_core/src/coin_select/coin_selector.rs

```diff
@@ -43,7 +43,8 @@ impl WeightedValue {
 #[derive(Debug, Clone, Copy)]
 pub struct CoinSelectorOpt {
     /// The value we need to select.
-    pub target_value: u64,
+    /// If the value is `None`, we drain everything.
+    pub target_value: Option<u64>,
```

Comment:
Isn't this comment wrong? The semantics should just be we don't check this constraint when we `.finish` if it's `None`.

Reply:
The way I am thinking about it is this:
* When we don't have a target value, we don't have recipient `txout`s. So then, we MUST have a drain output.
* Hence, for `finish` to succeed with no target value, we need to at least meet the minimum requirements for a drain `txout`.

Reply:
Maybe:
```
/// If the value is `None`, we send all non-fee amounts to the drain output.
```

Reply:
I'm not sure if we want to be that concrete.

```
if the value is `None` then the selection will be complete if it can pay for the drain output and satisfy the other constraints (e.g. minimum fees).
```

Not sure if I'm correct here though. Feel free to just merge this if you are happy with it and I'll take a closer look when I work on this bit.